core/test/utilities/helpers/navigation-helpers.spec.js

```diff
@@ -2,9 +2,106 @@
 const chai = require('chai');
 const assert = chai.assert;
 const sinon = require('sinon');
-import { NavigationHelpers } from '../../../src/utilities/helpers';
+import { AuthHelpers, NavigationHelpers } from '../../../src/utilities/helpers';
+import { LuigiAuth, LuigiConfig } from '../../../src/core-api';
 
 describe('Navigation-helpers', () => {
+  describe('isNodeAccessPermitted', () => {
+    let permissionCheckerFn;
+    beforeEach(() => {
+      permissionCheckerFn = sinon.spy();
+      sinon.stub(LuigiAuth, 'isAuthorizationEnabled');
+      sinon.stub(AuthHelpers, 'isLoggedIn');
+      sinon.stub(LuigiConfig, 'getConfigValue');
+    });
+    afterEach(() => {
+      permissionCheckerFn = undefined;
+      sinon.restore();
+    });
+
+    it('without permissionCheckerFn', () => {
+      LuigiConfig.getConfigValue.returns(null);
+      assert.isTrue(NavigationHelpers.isNodeAccessPermitted());
+    });
+
+    it('logged in, normal anonymousAccess', () => {
+      const checkNode = {
+        anonymousAccess: true
+      };
+
+      LuigiAuth.isAuthorizationEnabled.returns(true);
+      AuthHelpers.isLoggedIn.returns(true);
+
+      assert.isTrue(NavigationHelpers.isNodeAccessPermitted(checkNode));
+    });
+
+    it('logged out, normal anonymousAccess', () => {
+      const checkNode = {
+        anonymousAccess: true
+      };
+
+      LuigiAuth.isAuthorizationEnabled.returns(true);
+      AuthHelpers.isLoggedIn.returns(false);
+
+      assert.isTrue(NavigationHelpers.isNodeAccessPermitted(checkNode));
+    });
+
+    it('logged in, exclusive anonymousAccess', () => {
+      const checkNode = {
+        anonymousAccess: 'exclusive'
+      };
+
+      LuigiAuth.isAuthorizationEnabled.returns(true);
+      AuthHelpers.isLoggedIn.returns(true);
+      LuigiConfig.getConfigValue.returns(permissionCheckerFn);
+
+      assert.isFalse(NavigationHelpers.isNodeAccessPermitted(checkNode));
+    });
+
+    it('logged out, exclusive anonymousAccess, without permissionCheckerFn', () => {
+      const checkNode = {
+        anonymousAccess: 'exclusive'
+      };
+
+      LuigiAuth.isAuthorizationEnabled.returns(true);
+      AuthHelpers.isLoggedIn.returns(false);
+      LuigiConfig.getConfigValue.returns(undefined);
+
+      assert.isTrue(NavigationHelpers.isNodeAccessPermitted(checkNode));
+    });
+
+    it('logged out, exclusive anonymousAccess, with permissionCheckerFn', () => {
```

Comment:
This test is confusing IMHO since the mock of the function call returns `false` value, which is explicitly being returned above. It is also a little bit too much testing from my point of view. I would replace it with a test for the part where we are actually returning `false`, most precisely for the 2nd condition of the `if` block.

Reply:
makes sense. done